src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -429,18 +429,35 @@ public Response getVersion(@Context ContainerRequestContext crc,
            
             //If excludeFiles is null the default is to provide the files and because of this we need to check permissions. 
             boolean checkPerms = excludeFiles == null ? true : !excludeFiles;
+            
 
             Dataset dst = findDatasetOrDie(datasetId);
-            DatasetVersion dsv = getDatasetVersionOrDie(req, versionId, dst, uriInfo, headers, includeDeaccessioned, checkPerms);
-
-            if (dsv == null || dsv.getId() == null) {
+            DatasetVersion requestedDatasetVersion 
+                    = getDatasetVersionOrDie(req, versionId, dst, uriInfo, headers, includeDeaccessioned,
+                    checkPerms);
+            
+            DatasetVersion latestDatasetVersion = null;
+            boolean deaccesionedLookup = true;
+            //Check perms is false since we are never going to retrieve files, we are just getting the status of the version.
+            checkPerms = false;
+            if(versionId != DS_VERSION_LATEST){                
+                latestDatasetVersion = getDatasetVersionOrDie(req, DS_VERSION_LATEST, dst, uriInfo, headers,
+                        deaccesionedLookup,
+                        checkPerms);
+            } else {
+                latestDatasetVersion = requestedDatasetVersion;
+            }
+  
+            if (requestedDatasetVersion == null || requestedDatasetVersion.getId() == null) {
```

Comment:
I had to remove the condition because I found a bug when people would ask for the latest version specifically with the files depending on the permissions but besides that with the last update the code should work as intended, I added a few comments on the integration tests to explain the specific situations but I can explain more in detail, the logic is a bit complicated.